src/clr-angular/data/datagrid/providers/selection.ts

```diff
@@ -188,6 +188,29 @@ export class Selection {
         return false;
     }
 
+    /**
+     * Selects an item
+     */
+    private selectItem(item: any): void {
+        this.current.push(item);
+        if (this._items.trackBy) {
```

Comment:
@adityarb88 didn't you add an check on whether `this._items.all` is empty or not? Do we need that here?

Reply:
That one can't be called on initialization, so we're ok. The problem that Aditya fixed was when the selection was *initialized* before the items themselves were. Here, this method can only be called with an item that's being displayed, so the items have been initialized for sure.